ui-admin/src/study/surveys/CreateSurveyModal.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useContext, useState } from 'react'
+import { StudyEnvContextT } from 'study/StudyEnvironmentRouter'
+import Modal from 'react-bootstrap/Modal'
+import LoadingSpinner from 'util/LoadingSpinner'
+import Api, { VersionedForm } from 'api/api'
+import { useNavigate } from 'react-router-dom'
+import { Store } from 'react-notifications-component'
+import { failureNotification } from '../../util/notifications'
+import { PortalContext, PortalContextT } from '../../portal/PortalProvider'
+
+/** renders a modal that creates a new survey in a portal and configures it to the current study env */
+const CreateSurveyModal = ({ studyEnvContext, isReadOnlyEnv, show, setShow }: {studyEnvContext: StudyEnvContextT,
+  isReadOnlyEnv: boolean, show: boolean, setShow:  React.Dispatch<React.SetStateAction<boolean>> }) => {
+  const [isLoading, setIsLoading] = useState(false)
+  const [surveyName, setSurveyName] = useState('')
+  const [surveyStableId, setSurveyStableId] = useState('')
+
+  const portalContext = useContext(PortalContext) as PortalContextT
+  const navigate = useNavigate()
+
+  const createSurvey = async () => {
+    setIsLoading(true)
+    const createdSurvey = await Api.createNewSurvey(studyEnvContext.portal.shortcode,
+      {
+        createdAt: 0, id: '', lastUpdatedAt: 0, version: 1,
+        content: '{"pages":[]}', name: surveyName, stableId: surveyStableId
+      }).catch(e =>
+      Store.addNotification(failureNotification(`Error creating survey: ${e.message}`))
+    ) as VersionedForm
+
+    try {
+      await Api.createConfiguredSurvey(studyEnvContext.portal.shortcode,
```

Comment:
Should this be a separate API call or should it be combined with `createNewSurvey`? What happens if `createNewSurvey` succeeds but `createConfiguredSurvey` fails? I suspect it would put the application in a weird state that may not be easily represented to the user through the admin UI, which leads me to think that this should be a single transactional operation.

Reply:
Agreed that we may need #3 in the future. The downside of straddling #1 and #3 is that, if it ends up happening to someone that the operation only partially completes, it will require manual intervention to fix; they can't try again because the stable ID is already used, and they don't have a UI to do just the 2nd part themselves. The risk is probably small and it may never happen, but small risks like this will build up and incur real costs if we have too many of them. I think it's worth at least being aware of and deliberate about the accumulation of these small risks.